flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -381,11 +393,20 @@ public void onContainersAllocated(List<Container> containers) {
 					} catch (Throwable t) {
 						log.error("Could not start TaskManager in container {}.", container.getId(), t);
 
+						failedContainerSoFar.getAndAdd(1);
 						// release the failed container
 						workerNodeMap.remove(resourceId);
 						resourceManagerClient.releaseAssignedContainer(container.getId());
-						// and ask for a new one
-						requestYarnContainerIfRequired(container.getResource(), container.getPriority());
+
+						if (failedContainerSoFar.intValue() < maxFailedContainers) {
```

Comment:
I think we need to fail the cluster when failedContainerSoFar exceed the threshold after line 362 as well, right?

Reply:
Yes, you are right. Will change also.